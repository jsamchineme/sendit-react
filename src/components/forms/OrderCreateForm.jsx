import React from 'react'
import { Field, reduxForm } from 'redux-form';
import { orderCreateRules } from './rules/formRules';

export const renderField = ({
  input,
  label,
  type,
  className,
  placeholder,
  meta: { touched, error, warning }
}) => {
  let inputBox = <input className={className} {...input} placeholder={placeholder} type={type} />;
  if(type === 'textarea') {
    inputBox = <textarea className={className} {...input} placeholder={placeholder} />
  }
  return (
    <div className="item">
      <div className="field">{label}</div>
      {touched &&
        ((error && <div className="error-box active">{error}</div>))}
      <div className="value">
        {inputBox}
      </div>
    </div>
  )
}


export const OrderCreateForm = props => {
  const { handleSubmit, serverErrorMessage } = props;

  return (
    <form className="create-order-form" onSubmit={handleSubmit}>
      <div className="info-sections column col-7">
        <div className="item">
          {serverErrorMessage &&
          <div className="error-box active">{serverErrorMessage}</div>}
        </div>
        <Field
          name="from"
          type="text"
          className="line-input"
          component={renderField}
          label="Pickup Location"
          placeholder="Provide Pickup location"
          validate={orderCreateRules.from}
        />
        <Field
          name="to"
          type="text"
          className="line-input"
          component={renderField}
          label="Delivery Location"
          placeholder="Provide delivery location"
          validate={orderCreateRules.to}
        />
        <Field
          name="currentLocation"
          type="text"
          className="line-input"
          component={renderField}
          label="Present Location"
          placeholder="Provide your present location"
          validate={orderCreateRules.currentLocation}
        />
        <Field
          name="contactPhone"
          type="text"
          className="line-input"
          component={renderField}
          label="Receiver's phone"
          placeholder="Receiver's phone"
          validate={orderCreateRules.contactPhone}
        />
        <Field
          name="contactEmail"
          type="email"
          className="line-input"
          component={renderField}
          label="Receiver's email"
          placeholder="Receiver's email"
          validate={orderCreateRules.contactEmail}
        />
        <Field
          name="description"
          type="textarea"
          className="line-input"
          component={renderField}
          label="Parcel Description"
          placeholder="Provide brief description"
          validate={orderCreateRules.description}
        />
        <Field
          name="weight"
          type="text"
          className="line-input"
          component={renderField}
          label="Approximate weight of the parcel"
          placeholder="Provide the approximate weight of the parcel"
          validate={orderCreateRules.weight}
        />
        <div className="item actions">
          <button className="btn bg-light-orange medium-btn">Create Order</button>
        </div>
        <div className="images column col-5">
          <div className="image">
          </div>
        </div>
      </div>
    </form>
  )
}

const Form = reduxForm({
  form: 'orderCreate'
})(OrderCreateForm);

const OrderCreateFormContainer = (props) => {
  return <Form {...props} />
}

export default OrderCreateFormContainer;