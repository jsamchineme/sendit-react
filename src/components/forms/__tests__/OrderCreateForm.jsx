import React from 'react';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import { BrowserRouter, Route } from 'react-router-dom';
import OrderCreateForm from '../OrderCreateForm';
import configureStore from '../../../redux/configureStore';

const store = configureStore();

describe('Tests for login form', () => {
  it('should render the login form', () => {
    const wrapper = shallow(<OrderCreateForm />);
  
    // expect(wrapper.state('email')).toBe(value);
    expect(wrapper).toMatchSnapshot();
  });
});