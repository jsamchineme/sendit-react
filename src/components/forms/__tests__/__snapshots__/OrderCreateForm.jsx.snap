// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for login form should render the login form 1`] = `
<ReduxForm
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="orderCreate"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;
